### ISSUE TITLE
[Question] Building a Custom Sample

### ISSUE THREAD

**Issue by mohitsainiknl** [KhronosGroup/Vulkan-Samples#976]
I followed the provided build instructions and successfully obtained the `vulkan_samples.exe`, which can be utilized to run samples using the command `vulkan_samples.exe sample sample_id`.

For experimentation purposes, I created a new sample by using the following command:
```
./scripts/generate.py sample --name <SampleName> --category <category> --output-dir <output_dir>
```

However, I'm currently encountering difficulties in building this new sample. Could you please provide guidance on how to build and run custom samples?

Additionally, I'd like to understand the purpose of the `--data-path` option.

**Comment by SaschaWillems** [comment]
Can you elaborate? What difficulties do you encounter in building the new sample?

**Comment by mohitsainiknl** [comment]
Thank you for your response. I arrived here after exploring your "Vulkan C++ examples and demos" repository.

In typical CMake setups, we often utilize tools like `vcpkg` or `fetch_content` to incorporate external libraries. However, integrating the `vulkan-samples` repository into our projects doesn't seem as straightforward.

I'm currently facing challenges in leveraging this repository as an external dependency for my project. I haven't been able to find clear instructions on how to build a custom sample from it, especially when the path isn't within `samples` directory.

Could you please provide some guidance on this?

**Comment by SaschaWillems** [comment]
Integrating our samples into other repositories is not a use-case we currently support. Same for creating a sample outside of the `samples` directory. The script is mostly aimed at maintainers working on this repo. What is the actual use case of integrating our samples into another project, and why do you need to build samples outside of the `samples` folder?

**Comment by mohitsainiknl** [comment]
Thank you for your response.

Use Case:
I'm aiming to develop a rigging/animation tool, and I'm seeking a high-level API to work with. I prefer the flexibility to modify the API as needed, rather than starting directly with low-level APIs. After exploring options, I found `KhronosGroup/Vulkan-Samples` or `SaschaWillems/Vulkan` to be good starting points.

Challenges Faced:
1. Building a Custom Sample:
I created a sample using the command:
```
./scripts/generate.py sample --name my_project --category api
```
However, I'm encountering issues when attempting to build it. The command:
```
.\build\windows\app\bin\release\AMD64\vulkan_samples --data-path ./samples/api/my_project sample my_project
```
doesn't seem to work. I suspect I may need to register this new sample or use a different command.

2. Adding External Libraries:
I'm also unsure how to add external libraries (not available in `vulkan-samples`).

Please assist me, how can i address these challenges.

**Comment by SaschaWillems** [comment]
Sorry, I'm having a hard time following you.

> doesn't seem to work. I suspect I may need to register this new sample or use a different command.

What do you mean by "doesn't work"? Do you get an error message?

> I'm also unsure how to add external libraries (not available in vulkan-samples).

Also unsure what that means. Do you want to include the samples as an external library in your project? If so, that's nothing we support.

The `--data-path` argument can be used to specify a different asset path. You probably try to use it for something different.

As for creating a new sample and using it:

If you have run the script, run cmake again and the project will be build. 

E.g. after `python ./scripts/generate.py sample --name test --category api` you can see the name sample named test in the CMake setup:

![image](https://github.com/KhronosGroup/Vulkan-Samples/assets/10283127/04931c67-fd46-4cde-a5c9-f8bc04e33b19)

It's as simple as that.

**Comment by mohitsainiknl** [comment]
Apologies for the confusion.

> What do you mean by "doesn't work"? Do you get an error message?

I'm encountering the following errors:
![Screenshot (147)](https://github.com/KhronosGroup/Vulkan-Samples/assets/76866884/1314c5bf-93f0-4a37-b0d0-ac534c689ddf)

> Do you want to include the samples as an external library in your project?

No, my question is on, How to use external libraries (such as Eigen) within a sample that isn't included in the `vulkan-samples` repository. I realize this may not align with the repository's intended use.

I believe it would be best for me to focus on experimenting with the existing samples and take more time to understand the concepts at hand.

**Comment by SaschaWillems** [comment]
Did you run cmake and did you do a build after adding the new sample? The message in your screenshot suggests, that the new sample wasn't build.

As for your second question: That's totally out of scope for this repo.

**Comment by mohitsainiknl** [comment]
Yes, I did this:
```
cmake --build build/windows --config Release --target vulkan_samples
```
And, also build with `cmake-gui`:

![Screenshot (149)](https://github.com/KhronosGroup/Vulkan-Samples/assets/76866884/6bf4ccb5-d013-4ba4-979c-d4d2688f5483)

but, getting same error...

**Comment by tomadamatkinson** [comment]
Did you add the sample to the ORDERED_LIST in samples/CMakeLists.txt?

**Comment by mohitsainiknl** [comment]
Just added, but still getting same error.

**Comment by gpx1000** [comment]
Just a sanity check, after you generated your sample with the python script, did you build it with CMake?

You should have a cmake configure step AND a cmake build step.

**Comment by SaschaWillems** [comment]
> And, also build with `cmake-gui`:

You can't build with cmake-gui. As others noted, you need to actually build the samples with a compiler/IDE before you can use the new sample.

**Comment by mohitsainiknl** [comment]
**Yes, this worked for me.**

Here are the steps I followed:

1. I generated a new sample using the following command:
   ```
   python ./scripts/generate.py sample --name test --category api
   ```

2. I added the newly generated sample to the `ORDERED_LIST` in the `samples/CMakeLists.txt` file.

3. I configured the `Vulkan-Samples` project using the following CMake command:
   ```
   cmake -G "Visual Studio 17 2022" -A x64 -S . -Bbuild/windows
   ```

4. Next, I built the project using the following command:
   ```
   cmake --build build/windows --config Release --target vulkan_samples
   ```

5. Finally, I was able to run the sample using the command:
   ```
   .\build\windows\app\bin\release\AMD64\vulkan_samples sample test
   ```

@SaschaWillems @tomadamatkinson @gpx1000 

Thank you for the helpful guidance. Your assistance was invaluable.
Thanks again!
